static/js/src/advantage/subscribe/react/APICalls/PendingPurchase.jsx

```diff
@@ -0,0 +1,77 @@
+import { useStripe } from "@stripe/react-stripe-js";
+import { useState } from "react";
+import { useQuery } from "react-query";
+import { getPurchase, postInvoiceID } from "../../../contracts-api";
+
+const usePendingPurchase = () => {
+  const [pendingPurchaseID, setPendingPurchaseID] = useState();
+
+  const stripe = useStripe();
+
+  const { isLoading, isError, isSuccess, data, error } = useQuery(
+    "pendingPurchase",
+    async () => {
+      const res = await getPurchase(pendingPurchaseID);
+
+      if (!res.stripeInvoices) {
+        throw new Error("Missing invoice");
```

Comment:
There is no need to throw here, please handle the error gracefully instead of panicking

Reply:
It's the way `react-query` handles errors https://react-query.tanstack.com/guides/query-functions#handling-and-throwing-errors